src/components/parameter-pages/type-filter-table.tsx

```diff
@@ -0,0 +1,114 @@
+import { activeTypeFilterParamsState, TypeFilter } from '@/state/type-filter-params-state';
+import {
+  Button,
+  Checkbox,
+  CloseButton,
+  Table,
+  TableCaption,
+  TableTbody,
+  TableTd,
+  TableTh,
+  TableThead,
+  Text,
+  TableTr,
+  ActionIcon,
+  Tooltip,
+  Flex
+} from '@mantine/core';
+import { IconCopy } from '@tabler/icons-react';
+import { Dispatch, SetStateAction } from 'react';
+import { useRecoilState } from 'recoil';
+
+export default function TypeFilterTable({ setFilterInput }: { setFilterInput: Dispatch<SetStateAction<string>> }) {
+  const [typeFilters, setTypeFilters] = useRecoilState(activeTypeFilterParamsState);
+
+  const toggleCheckbox = (thisTypeFilter: TypeFilter) => {
+    const toggledList = typeFilters.map(typeFilter => {
+      return {
+        filter: typeFilter.filter,
+        active: typeFilter.filter === thisTypeFilter.filter ? !typeFilter.active : typeFilter.active
+      } as TypeFilter;
+    });
+
+    setTypeFilters(toggledList);
+  };
+
+  const toggleAll = () => {
+    if (typeFilters.filter(typeFilter => !typeFilter.active).length === 0)
+      setTypeFilters(
+        typeFilters.map(typeFilter => {
+          return { filter: typeFilter.filter, active: false };
+        })
+      );
+    else
+      setTypeFilters(
+        typeFilters.map(typeFilter => {
+          return { filter: typeFilter.filter, active: true };
+        })
+      );
+  };
+
+  return (
+    <Table withTableBorder>
+      <TableThead bg="gray.2">
+        <TableTh>
+          <Checkbox
+            checked={typeFilters.filter(val => !val.active).length === 0 && typeFilters.length !== 0}
```

Comment:
```suggestion
            checked={typeFilters.filter(val => !val.active).length === 0 && typeFilters.length > 0}
```
Just for consistency